test/index.js

```diff
@@ -55,3 +61,18 @@ test('type', function() {
   eq(type(new Number(0)), 'Number');
   eq(type(new String('')), 'String');
 });
+
+test('parse', function() {
+  eq(parse('package/Type'), TypeIdentifier('package', 'Type', 0));
+  eq(parse('package/Type/X'), TypeIdentifier('package/Type', 'X', 0));
+  eq(parse('@scope/package/Type'), TypeIdentifier('@scope/package', 'Type', 0));
+  eq(parse(''), TypeIdentifier(null, '', 0));
+  eq(parse('/Type'), TypeIdentifier(null, '/Type', 0));
+  eq(parse('@0'), TypeIdentifier(null, '@0', 0));
+  eq(parse('foo/\n@1'), TypeIdentifier('foo', '\n', 1));
+  eq(parse('Type@1'), TypeIdentifier(null, 'Type@1', 0));
+  eq(parse('package/Type@1'), TypeIdentifier('package', 'Type', 1));
+  eq(parse('package/Type@999'), TypeIdentifier('package', 'Type', 999));
+  eq(parse('package/Type@X'), TypeIdentifier('package', 'Type@X', 0));
+  eq(parse('package////@3@2@1@1'), TypeIdentifier('package///', '@3@2@1', 1));
+});
```

Comment:
You've done a nice job with these test cases. Each test case exists to test a specific pathway. :thumbsup: